### ISSUE TITLE
A way to create HTML tables.

### ISSUE THREAD

**Issue by Brahvim** [processing/p5.js#5154]
#### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [x] DOM
- [ ] Events
- [ ] Friendly error system
- [ ] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### New feature details:
Basically, this is a request for something which would insert a table element in the HTML file, to help with making tables; one would be saved from the hassle of typing every tag to make an HTML table, and it would ALSO be possible to make tables generate with data that was generated procedurally instead of being hardcoded into the HTML, etc.

**Comment by Prateek93a** [comment]
@Brahvim This might be helpful : https://p5js.org/reference/#/p5.Table

**Comment by Brahvim** [comment]
> @Brahvim This might be helpful : https://p5js.org/reference/#/p5.Table 

Wait, but does it work with HTML? Because what I requested was for it to work like a DOM element.

The reference page itself does not mention that it is a DOM element anywhere. 
(_Unless I misread!_)

Thank you for the information though! :D!

**Comment by Prateek93a** [comment]
Oh ok, I misread your request and thought you were asking for tables feature in p5 sketch. Thanks for the clarification!

**Comment by outofambit** [comment]
Hey @Brahvim, thanks for filing this request! Did you have any thoughts on how would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to p5.js? This seems like an interesting idea (and the DOM module is one of my favorite parts of p5.js!) but I'm having trouble connecting this idea to [our focus on access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md). What do you think?

**Comment by Brahvim** [comment]
> Hey @Brahvim, thanks for filing this request! Did you have any thoughts on how would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to p5.js? This seems like an interesting idea (and the DOM module is one of my favorite parts of p5.js!) but I'm having trouble connecting this idea to [our focus on access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md). What do you think?

Uh-oh!~ Looks like I used that tag wrongly! Didn't really know what it meant. I am sorry, I will remove it if I can. Thank you for addressing the mistake.

**Comment by micuat** [comment]
@outofambit just a thought but can this be helpful for listing descriptions for screen readers? In fact the current implementation of `describeElement` generates a table in a naive way, but this could be wrapped by `createTable` if that is implemented as part of dom library
https://github.com/processing/p5.js/blob/main/src/accessibility/describe.js#L174

**Comment by Brahvim** [comment]
> @outofambit just a thought but can this be helpful for listing descriptions for screen readers? In fact the current implementation of `describeElement` generates a table in a naive way, but this could be wrapped by `createTable` if that is implemented as part of dom library
> https://github.com/processing/p5.js/blob/main/src/accessibility/describe.js#L174

*Good idea,* thank you!

**Comment by Brahvim** [comment]
I recently found out about the p5 `saveTable()` method. Maybe it is possible to generate an HTML table with it and add that to the HTML file? I think that is possible too. Might not need to keep this issue open if that is possible to do.

**Comment by micuat** [comment]
@Brahvim it's close but not exactly what you want - it's for outputting a file instead of DOM
https://github.com/processing/p5.js/blob/374acfb44588bfd565c54d61264df197d798d121/src/io/files.js#L1747

**Comment by Brahvim** [comment]
Yes, I know that, and that's what I said - since this generates markup for a table, why not find/implement a way to insert that code into the ```index.html``` file? Given an ```id``` attribute, it'll then be very easy to apply style information, etc.

**Comment by micuat** [comment]
@Brahvim I get your point but it sounds a bit like hotglue / adhoc solution to me. If this is going to be implemented, I think it's fair to make a new "class" like `TableElement` to manage it because the existing solutions for `describeElement` and `saveTable` are specific to output a table and, for example, you cannot simply add a row/column/cell.

There is also a problem with `p5.createElement("td")` that the performance is not optimal for a big table (the OP says >2000, but I see huge performance drop even with ~200)
https://discourse.processing.org/t/avoid-csv-overlaps-for-website-project/29723/11

But again, we need to justify if this is needed from the perspective of accessibility (otherwise we end up adding every possible feature to p5.js). You can always use other libraries to generate a table.

**Comment by Brahvim** [comment]
> @Brahvim I get your point but it sounds a bit like hotglue / adhoc solution to me. If this is going to be implemented, I think it's fair to make a new "class" like `TableElement` to manage it because the existing solutions for `describeElement` and `saveTable` are specific to output a table and, for example, you cannot simply add a row/column/cell.
> 
> There is also a problem with `p5.createElement("td")` that the performance is not optimal for a big table (the OP says >2000, but I see huge performance drop even with ~200)
> https://discourse.processing.org/t/avoid-csv-overlaps-for-website-project/29723/11
> 
> But again, we need to justify if this is needed from the perspective of accessibility (otherwise we end up adding every possible feature to p5.js). You can always use other libraries to generate a table.

I see. Should probably leave it up to the developers of p5 now...?

**Comment by Brahvim** [comment]
Since it ended up seeming like a performance issue to do this... I don't think it's a good idea to add something like this to p5. Feel free to try though :)!  

If your solution fits p5's style of abstractions and offers a lot of **convenience**, it'd be great to have :D!  

In my opinion it is still better to add elements using the `document` object directly and have a good understanding of Web APIs. But if you're up for the challenge of making something that suits p5's style, sure do take it on :)!

**Comment by Mamatha1718** [comment]
Hi @Brahvim , 
Thank you for your detailed response and for clarifying the challenges associated with implementing this feature. I appreciate the insights on performance concerns, alignment with p5.js's philosophy, and the existing alternatives like createElement() or vanilla JavaScript.

I understand that this feature might not be a priority at the moment, but I’m still eager to contribute to the p5.js community and learn more about its development process. Are there any other open issues or areas where I could focus my efforts? I’m particularly interested in learning more about Embedding Methods performance optimization, or creating new features that align with p5.js's goals.

Looking forward to your suggestions!
Thank You in Advance
